packages/pods-core/src/basev3.ts

```diff
@@ -22,7 +22,16 @@ export enum PROMPT {
 export type PodOpts<T> = {
   engine: DEngineClient;
   config: T;
-  onPrompt?: (arg0?: PROMPT) => Promise<any | undefined>
+  onPrompt?: (arg0?: PROMPT) => Promise<any | undefined>;
+  utilityMethods?: {
+    getHierarchyDest: (arg0: string) => Promise<string | undefined>;
```

Comment:
I understand that you introduced lambda's here for testability, but are all of them necessary to expose for the tests to run? Can some of these be stubbed within the test?

Reply:
Apart from testability, I added these for type declaration of all the methods in `utilityMethods`.